concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -380,9 +382,13 @@ loadSkovPersistentData rp _treeStateDirectory pbsc = do
   (_lastFinalizationRecord, lfStoredBlock) <- liftIO (getLastBlock _db) >>= \case
       Left s -> logExceptionAndThrowTS $ DatabaseInvariantViolation s
       Right r -> return r
-  _lastFinalized <- liftIO (makeBlockPointer lfStoredBlock)
-
-  return SkovPersistentData {
+  elfBlob <- liftIO . try $
+    readBlobBSFromHandle (bscBlobStore . PBS.pbscBlobStore $ pbsc) (sbState lfStoredBlock)
```

Comment:
It would be very good to document here why this check is enough.

it relies on the fact that the reference is written after the data.